### PR TITLE
infra(unicorn): prefer-array-some

### PR DESCRIPTION
**Pull request by ST-DDT**
Ref: #2439

- #2439

---

Enables the [`unicorn/prefer-array-some`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-array-some.md) lint rule.

### CODE REVIEW

File: test/modules/system.spec.ts [hunk 1]
```diff
@@ -405,13 +405,11 @@ describe('system', () => {
         );
 
         it('should be able to return non-standard cron expressions', () => {
-          const validResults = [...'0123456789'.split(''), '*', '@'];
+          const validResults = new Set('0123456789*@');
           expect(
             faker.system.cron({ includeNonStandard: true })[0],
```
Comment (matthewmayer) [inline test/modules/system.spec.ts]: This whole test seems awkwardly written. If we're only testing the first char maybe refactor to use startsWith?

**Review by matthewmayer — CHANGES_REQUESTED** [review]

File: test/modules/system.spec.ts (hunk 1 above)
Comment (ST-DDT) [reply test/modules/system.spec.ts]: I'm not sure how to make it less awkward.

Maybe like this:

````ts
const validPrefixes = [...'0123456789*@'];
const actual = faker.system.cron({ includeNonStandard: true });
expect(
    validPrefixes.some((prefix) => actual.startsWith(prefix))
).toBe(true);
````

?

Comment (ST-DDT) [reply test/modules/system.spec.ts]: Alternatively:

````ts
const validResults = new Set('0123456789*@');
const actual = faker.system.cron({ includeNonStandard: true })[0];

expect(validResults).toContain(actual);
````

Comment (ST-DDT) [reply test/modules/system.spec.ts]: Done

**Review by xDivisionByZerox — DISMISSED** [review]

**Review by ejcheng — DISMISSED** [review]

**Review by matthewmayer — DISMISSED** [review]

**Comment by ST-DDT** [comment]
Ready for review

**Review by xDivisionByZerox — APPROVED** [review]

**Pull request merged**